examples/utils.py

```diff
@@ -108,9 +114,10 @@ def train_with_torch(
             optimizer.step()
         loss = postproc(step, loss) if postproc is not None else loss
 
-        if step % 10 == 0:
-            logger.info(f"Step {step}, LOSS: {loss.item()}", ranks=0)
+        global_step = step // micro_batch_size
```

Comment:
The `step` is the index of the micro_batch, so `step // (global_batch_size // micro_batch_size)` is the right way to count the global step. Another way is leverage the runtime: 
```python
        if model.global_steps % 10 == 0 and model.is_gradient_accumulation_boundary():
            logger.info(f"Step {model.global_steps}, LOSS: {loss.item()}", ranks=0)

        loss = None
        if model.global_steps == steps:
            break
```